src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -257,59 +251,60 @@ void CoreWorkerDirectActorTaskReceiver::HandleDirectActorAssignTask(
+                    auto num_returns = task_spec.NumReturns();
+                    RAY_CHECK(num_returns > 0);
+                    if (task_spec.IsActorCreationTask() || task_spec.IsActorTask()) {
+                      // Decrease to account for the dummy object id.
+                      num_returns--;
+                    }
+
+                    // TODO(edoakes): resource IDs are currently kept track of in the
+                    // raylet, need to come up with a solution for this.
+                    ResourceMappingType resource_ids;
+                    std::vector<std::shared_ptr<RayObject>> return_objects;
+                    auto status = task_handler_(task_spec, resource_ids, &return_objects);
+                    if (status.IsSystemExit()) {
+                      // In Python, SystemExit can only be raised on the main thread. To
+                      // work around this when we are executing tasks on worker threads,
+                      // we re-post the exit event explicitly on the main thread.
+                      task_main_io_service_.post([this]() { exit_handler_(); });
+                      return;
+                    }
+                    RAY_CHECK(return_objects.size() == num_returns)
+                        << return_objects.size() << "  " << num_returns;
+
+                    for (size_t i = 0; i < return_objects.size(); i++) {
+                      auto return_object = reply->add_return_objects();
+                      ObjectID id = ObjectID::ForTaskReturn(
+                          task_spec.TaskId(), /*index=*/i + 1,
+                          /*transport_type=*/static_cast<int>(TaskTransportType::DIRECT));
+                      return_object->set_object_id(id.Binary());
+
+                      // The object is nullptr if it already existed in the object store.
+                      const auto &result = return_objects[i];
+                      if (result == nullptr || result->GetData()->IsPlasmaBuffer()) {
+                        return_object->set_in_plasma(true);
+                      } else {
+                        if (result->GetData() != nullptr) {
+                          return_object->set_data(result->GetData()->Data(),
+                                                  result->GetData()->Size());
+                        }
+                        if (result->GetMetadata() != nullptr) {
+                          return_object->set_metadata(result->GetMetadata()->Data(),
+                                                      result->GetMetadata()->Size());
+                        }
+                      }
+                    }
+
+                    send_reply_callback(status, nullptr, nullptr);
+                  },
+                  [send_reply_callback]() {
+                    send_reply_callback(Status::Invalid("client cancelled rpc"), nullptr,
+                                        nullptr);
+                  },
+                  dependencies);
```

Comment:
are there any actual changes in this if block, or it's just indent?

Reply:
Just formatting.